sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/BaseTypeTracker.cs

```diff
@@ -112,8 +113,24 @@ bool IsTrackedRelationship(SyntaxNode contextNode, SemanticModel semanticModel,
                 issueLocation = null;
                 return false;
             };
-        
-        
+
+        internal BaseClassCondition WhenDerivesFrom(ImmutableArray<KnownType> types) =>
+            (BaseTypeContext context, out Location issueLocation) =>
+            {
+                foreach (var baseTypeNode in context.AllBaseTypeNodes)
+                {
+                    if (context.Model.GetTypeInfo(baseTypeNode).Type?.DerivesFromAny(types) ?? false)
```

Comment:
Perhaps this does not matter much with the current rule, but I think it makes sense to cache the results of GetTypeInfo in Lazy or something else, so that we don't obtain the symbol for each of the checked types